algo/src/test/java/com/hartwig/serve/sources/ckb/CkbKnownEventsExtractorTest.java

```diff
@@ -0,0 +1,131 @@
+        extractedEvents.add(createExon("APC", "exon 3 deletion", "unknown"));
+        extractedEvents.add(createFusion("ALK", "EML4", "gain of function"));
+        extractedEvents.add(createFusion("APC", "BRAF", "unknown"));
+        extractedEvents.add(createHotspot("BRAF", "V600E", "gain of function"));
+        extractedEvents.add(createHotspot("TP53", "R175H", "unknown"));
+        extractedEvents.add(createAmplification("MET", "gain of function"));
+        extractedEvents.add(createAmplification("KIT", "unknown"));
+
+        KnownEvents knownEvents = CkbKnownEventsExtractor.generateKnownEvents(extractedEvents, false);
+
+        assertEquals(1, knownEvents.hotspots().size());
+        assertEquals(ProteinEffect.GAIN_OF_FUNCTION, knownEvents.hotspots().iterator().next().proteinEffect());
+
+        assertEquals(1, knownEvents.copyNumbers().size());
+        assertEquals(ProteinEffect.GAIN_OF_FUNCTION, knownEvents.copyNumbers().iterator().next().proteinEffect());
+
+        assertEquals(1, knownEvents.fusions().size());
+        assertEquals(ProteinEffect.GAIN_OF_FUNCTION, knownEvents.fusions().iterator().next().proteinEffect());
+
+        assertEquals(1, knownEvents.exons().size());
+        assertEquals(ProteinEffect.LOSS_OF_FUNCTION, knownEvents.exons().iterator().next().proteinEffect());
+
+        assertEquals(1, knownEvents.codons().size());
+        assertEquals(ProteinEffect.GAIN_OF_FUNCTION, knownEvents.codons().iterator().next().proteinEffect());
+    }
+
+    @NotNull
+    private static ExtractedEvent createCodon(@NotNull String gene, @NotNull String codon, @Nullable String proteinEffect) {
+        return ImmutableExtractedEvent.builder()
+                .gene(gene)
+                .event(codon)
+                .variant(CkbTestFactory.createVariant(gene, codon, gene + " " + codon, proteinEffect))
+                .eventType(EventType.CODON)
+                .eventExtractorOutput(ImmutableEventExtractorOutput.builder()
+                        .addCodons(ImmutableCodonAnnotation.builder()
+                                .from(RangeTestFactory.createRangeAnnotation(gene, "1", 1, 1, MutationType.INFRAME_DELETION))
+                                .inputTranscript("transcript")
+                                .inputCodonRank(1)
+                                .build())
+                        .build())
+                .build();
+    }
+
+    @NotNull
+    private static ExtractedEvent createExon(@NotNull String gene, @NotNull String event, @Nullable String proteinEffect) {
+        return ImmutableExtractedEvent.builder()
+                .gene(gene)
+                .event(event)
+                .variant(CkbTestFactory.createVariant(gene, event, gene + " " + event, proteinEffect))
+                .eventType(EventType.EXON)
+                .eventExtractorOutput(ImmutableEventExtractorOutput.builder()
+                        .addExons(ImmutableExonAnnotation.builder()
```

Comment:
I would use `RangeTestFactory.knownExonBuilder` here, that way you don't have to specify input transcript and input exon rank, since they don't matter for your test